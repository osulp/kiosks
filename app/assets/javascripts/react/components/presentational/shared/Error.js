import { connect } from 'react-redux';
import React, { Component, PropTypes } from 'react';

class Error extends Component {
  render() {
    return (
      <div id="error">
        {this.props.errors.map((e, i, a) => { return (<span key={`error.${i}`}>{e.message}</span>)})}
      </div>
    );
  }
}

Error.propTypes = {
  errors: PropTypes.array,
};

export default Error;
