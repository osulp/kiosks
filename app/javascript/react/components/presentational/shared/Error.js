import { connect } from "react-redux"
import React, { Component } from "react"
import PropTypes from "prop-types"

class Error extends Component {
  render() {
    return (
      <div id="error">
        {this.props.errors.map((e, i, a) => {
          return <span key={`error.${i}`}>{e.message}</span>
        })}
      </div>
    )
  }
}

Error.propTypes = {
  errors: PropTypes.array
}

export default Error
