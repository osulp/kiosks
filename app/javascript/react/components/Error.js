import { bindActionCreators } from "redux"
import { connect } from "react-redux"

import Error from "./presentational/shared/Error"
import * as actions from "../actions"

// Top level mapping of the application state to properties of component that is being
// connected.
export const mapStateToProps = state => {
  return {
    errors: state.kiosk.errors
  }
}

// Map the actions to the dispatcher
export const mapDispatchToProps = dispatch => {
  return bindActionCreators(actions, dispatch)
}

// Connect the mappings (state -> properties, and actions -> dispatch) to the component
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Error)
