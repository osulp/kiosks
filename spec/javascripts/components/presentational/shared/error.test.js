import React from 'react';
import Error from '../../../../../app/assets/javascripts/react/components/presentational/shared/Error';
import * as factories from '../../../.factories';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

describe('Snapshot', () => {

  const props = {
    errors: [factories.error],
    fetchSlides: jest.fn()
  };

  const component = renderer.create(
    <Error {...props} />
  );

  it('should match the cached snapshot', () => {
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
