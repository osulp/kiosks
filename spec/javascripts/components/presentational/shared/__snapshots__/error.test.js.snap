exports[`Snapshot should match the cached snapshot 1`] = `
<div
  id="error">
  <span>
    Error
  </span>
</div>
`;
